### PR TITLE
PR for create post, edit and delete post with post context

### PR DESCRIPTION
**Pull request by manismk**
- Added create, edit post modal
- handle create, edit, and delete post for firebase
- [Preview](https://socialoo-dev.netlify.app/login)

### CODE REVIEW

File: src/components/createPostModal/CreatepostModal.jsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+              className="btn btn--primary m-r-1"
+              onClick={closeModal}
+            >
+              Cancel
+            </button>
+
+            <button
+              href="#"
+              className="btn btn--primary "
+              onClick={clickHandler}
```
Comment (shraddha15-creator) [inline src/components/createPostModal/CreatepostModal.jsx:155]: you can give a better name to this clickHandler function. Like if it's creating a post then the name can be `createPostHandler`.

**Review by shraddha15-creator — APPROVED** [review]
Very nice and clean UI
Nice color theme. 
Code is clean and readable

File: src/components/createPostModal/CreatepostModal.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  const { allUsers } = useUser();
+
+  useEffect(() => {
+    if (posts.isFromEdit) {
+      if (posts.editData.imageUrl === null) {
+        setPostData((prev) => ({ ...prev, caption: posts.editData.caption }));
+      } else {
+        setPostData((prev) => ({
+          ...prev,
+          caption: posts.editData.caption,
```
Comment (xsaurabhsuthar) [inline src/components/createPostModal/CreatepostModal.jsx:27]: yout can destructure it and then pass as it is.

File: src/components/createPostModal/CreatepostModal.jsx [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+      imageLink: URL.createObjectURL(file),
+      imageFileError: "",
+    }));
+  };
+
+  const clickHandler = () => {
+    if (postData.caption.length && postData.imageFileError.length === 0) {
+      if (posts.isFromEdit) {
+        handleEditPost(
+          posts.editData,
```
Comment (xsaurabhsuthar) [inline src/components/createPostModal/CreatepostModal.jsx:48]: suggestion: you can destructure and pass it.

**Review by xsaurabhsuthar — APPROVED** [review]

File: src/components/createPostModal/CreatepostModal.jsx (hunk 2 above)
Comment (manismk) [reply src/components/createPostModal/CreatepostModal.jsx:27]: Thanks for review, will do that

**Pull request merged**